src/OpenSleigh.Transport.RabbitMQ/PublisherChannelContextPool.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Concurrent;
+using RabbitMQ.Client;
+
+namespace OpenSleigh.Transport.RabbitMQ
+{
+    internal sealed class PublisherChannelContextPool : IPublisherChannelContextPool, IDisposable
+    {
+        private readonly IBusConnection _connection;
+        private readonly ConcurrentDictionary<string, ConcurrentBag<PublisherChannelContext>> _pools = new ();
+        
+        public PublisherChannelContextPool(IBusConnection connection)
+        {
+            _connection = connection ?? throw new ArgumentNullException(nameof(connection));
+        }
+
+        public PublisherChannelContext Get(QueueReferences references)
+        {
+            if (references == null) 
+                throw new ArgumentNullException(nameof(references));
+
+            var pool = _pools.GetOrAdd(references.ExchangeName, _ => new());
+            
+            if (!pool.TryTake(out var ctx))
+            {
+                var channel = _connection.CreateChannel();
+                channel.ExchangeDeclare(exchange: references.ExchangeName, type: ExchangeType.Fanout);
+                ctx = new PublisherChannelContext(channel, references, this);
+            }
+
+            return ctx;
+        }
+
+        public void Return(PublisherChannelContext ctx)
+        {
+            if (ctx == null) 
+                throw new ArgumentNullException(nameof(ctx));
+            
+            var pool = _pools.GetOrAdd(ctx.QueueReferences.ExchangeName, _ => new());
+            pool.Add(ctx);
```

Comment:
before adding back to the pool, you have to be sure that the channel it's not in fault, I don't know if IModel gives you this ability, for sure the channel is closed if something goes wrong on rabbitmq server side.

Reply:
that was not meant to be there, I was just testing something and forgot to remove it. Good catch!

Reply:
good point, I'll put a guard around it.